server/BUILD

```diff
@@ -51,7 +76,11 @@ rust_library(
         "@crates//:tracing",
         "@crates//:uuid",
         "@crates//:pwhash",
-    ]
+    ],
+    crate_features = select({
+        ":development_mode": [],
+        ":published_mode": ["published"],
```

Comment:
Cargo sync is fine. 
`bazel run //:typedb_server_bin --//server:mode=published` runs the server in the published mode (not "release" so we don't get confused by the code performance without optimizations)
`bazel run //:typedb_server_bin --//server:mode=development` and `bazel run //:typedb_server_bin` run the server in the development mode (no `published` crate feature)

The cargo sync tool doesn't add this feature by default. Thus, `cargo build / run` work with development mode by default for the local builds, which is what we needed. However, we can either add the `published` feature to our `Cargo.toml`s or just use the CLI switch for the development mode if needed.

Reply:
that's fine!